Sources/Plasma/Apps/plClient/Mac-Cocoa/PLSPatcher.mm

```diff
@@ -139,13 +194,145 @@ bool IApproveDownload(const plFileName& file)
     return extExcludeList.find(file.GetFileExt()) == extExcludeList.end();
 }
 
+static la_ssize_t copy_data(struct archive *ar, struct archive *aw)
+{
+    while (true) {
+        la_ssize_t r;
+        const void *buff;
+        size_t size;
+        la_int64_t offset;
+        
+        r = archive_read_data_block(ar, &buff, &size, &offset);
+        if (r == ARCHIVE_EOF)
+            return (ARCHIVE_OK);
+        if (r < ARCHIVE_OK)
+            return (r);
+        r = archive_write_data_block(aw, buff, size, offset);
+        if (r < ARCHIVE_OK) {
+            fprintf(stderr, "%s\n", archive_error_string(aw));
+            return (r);
+        }
+    }
+}
+
+void Patcher::ISelfPatch(const plFileName& file)
+{
+    /*
+     Note on errors:
+     This function does not return errors, but a self patch
+     without a populated updatedClientURL will imply something
+     went wrong during decompress.
+     */
+    
+    PLSPatcher* patcher = parent;
+    patcher.selfPatched = true;
+    
+    int flags;
+    la_ssize_t r;
+
+    /* Select which attributes we want to restore. */
+    flags = ARCHIVE_EXTRACT_TIME;
+    
+    struct archive *a = archive_read_new();
+    struct archive *ext = archive_write_disk_new();
+    
+    {
+        int error;
+        error = archive_read_support_format_tar(a);
+        hsAssert(error == ARCHIVE_OK, "Unable to set tar format option");
+        error = archive_read_support_filter_gzip(a);
+        hsAssert(error == ARCHIVE_OK, "Unable to set gzip filter");
+        error = archive_read_support_filter_bzip2(a);
+        hsAssert(error == ARCHIVE_OK, "Unable to set bzip filter");
+        
+        error = archive_write_disk_set_options(ext, flags);
```

Comment:
If this is the only use of the `flags` variable, could just move the `ARCHIVE_EXTRACT_TIME` in-line here (I realize previously there were multiple flags being used so it made sense to have a variable)